experimental/packages/opentelemetry-sdk-metrics-base/src/state/TemporalMetricProcessor.ts

```diff
@@ -152,19 +145,45 @@ export class TemporalMetricProcessor<T> {
     return result;
   }
 
-  static merge<T>(last: AttributeHashMap<T>, current: AttributeHashMap<T>, aggregator: Aggregator<T>) {
+  static merge<T extends Maybe<Accumulation>>(last: AttributeHashMap<T>, current: AttributeHashMap<T>, aggregator: Aggregator<T>) {
     const result = last;
     const iterator = current.entries();
     let next = iterator.next();
     while (next.done !== true) {
       const [key, record, hash] = next.value;
-      const lastAccumulation = last.get(key, hash) ?? aggregator.createAccumulation();
-      result.set(key, aggregator.merge(lastAccumulation, record), hash);
+      if (last.has(key, hash)) {
+        const lastAccumulation = last.get(key, hash);
+        // lastAccumulation must present.
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const accumulation = aggregator.merge(lastAccumulation!, record);
+        result.set(key, accumulation, hash);
+      } else {
+        result.set(key, record, hash);
+      }
```

Comment:
IMO readability is improved without `has/get` and just using `get` and `if`. Is this a performance optimization to avoid the allocation for `lastAccumulation` in cases where it is not there? I think it should be present in most instances right?

```suggestion
      const lastAccumulation = last.get(key, hash);
      if (lastAccumulation) {
        result.set(key, aggregator.merge(lastAccumulation, record), hash);
      } else {
        result.set(key, record, hash);
      }
```

Reply:
The accumulation value can be `undefined` for DropAggregator (or maybe we should change it to be `null` to better reflect the intention?). So we have to check the existence of an attribute key instead of checking the nullish of value.